raft/fsm.cc

```diff
@@ -41,10 +41,6 @@ fsm::fsm(server_id id, term_t current_term, server_id voted_for, log log,
     }
 }
 
-fsm::fsm(server_id id, term_t current_term, server_id voted_for, log log,
```

Comment:
I am not sure there is a merit to this patch. persisting commit index it optional. We happen to do it in Scylla and this is the only non test user, but the functionality is general.

Reply:
Easier to make a conscious decision "I don't want to restore commit index" by explicitly providing `index_t{0}` in constructor. Minor nuisance but always better to make someone think for a second than not IMO.

Also fewer constructors to extend when adding new stuff like I do in later commit.

Reply:
OK, but IMO the problem with requiring explicit `index_t{0}` is that it is not clear from the API that it is optional.  How a user of the API knows it can provide  `index_t{0}`  and not the real value and the code will work?

Reply:
You're pointing to a larger problem here --- the fact that `fsm` constructor is not documented at all.
I can fill this hole.

Reply:
Documentation is good. Making signature more descriptive is also good (does not obliterate the need for the documentation of course). Making it `std::optional` would be more descriptive, but OTH not sure it is worth it.